Client/Writes/PointData.Builder.cs

```diff
@@ -0,0 +1,300 @@
+            public Builder Field(string name, ulong value)
+            {
+                return PutField(name, value);
+            }
+
+            /// <summary>
+            /// Add a field with a <see cref="uint"/> value.
+            /// </summary>
+            /// <param name="name">the field name</param>
+            /// <param name="value">the field value</param>
+            /// <returns>this</returns>
+            public Builder Field(string name, uint value)
+            {
+                return PutField(name, value);
+            }
+
+            /// <summary>
+            /// Add a field with a <see cref="string"/> value.
+            /// </summary>
+            /// <param name="name">the field name</param>
+            /// <param name="value">the field value</param>
+            /// <returns>this</returns>
+            public Builder Field(string name, string value)
+            {
+                return PutField(name, value);
+            }
+
+            /// <summary>
+            /// Add a field with a <see cref="bool"/> value.
+            /// </summary>
+            /// <param name="name">the field name</param>
+            /// <param name="value">the field value</param>
+            /// <returns>this</returns>
+            public Builder Field(string name, bool value)
+            {
+                return PutField(name, value);
+            }
+
+            /// <summary>
+            /// Updates the timestamp for the point.
+            /// </summary>
+            /// <param name="timestamp">the timestamp</param>
+            /// <param name="timeUnit">the timestamp precision</param>
+            /// <returns></returns>
+            public Builder Timestamp(long timestamp, WritePrecision timeUnit)
+            {
+                _precision = timeUnit;
+                _time= timestamp;
+                return this;
+            }
+
+            /// <summary>
+            /// Updates the timestamp for the point represented by <see cref="TimeSpan"/>.
+            /// </summary>
+            /// <param name="timestamp">the timestamp</param>
+            /// <param name="timeUnit">the timestamp precision</param>
+            /// <returns></returns>
+            public Builder Timestamp(TimeSpan timestamp, WritePrecision timeUnit)
+            {
+                BigInteger? time = null;
+                switch (timeUnit)
+                {
+                    case WritePrecision.Ns:
+                        time = timestamp.Ticks * 100;
+                        break;
+                    case WritePrecision.Us:
+                        time = (BigInteger)(timestamp.Ticks * 0.1);
+                        break;
+                    case WritePrecision.Ms:
+                        time = (BigInteger)timestamp.TotalMilliseconds;
+                        break;
+                    case WritePrecision.S:
+                        time = (BigInteger)timestamp.TotalSeconds;
+                        break;
+                }

```

Comment:
Could you please reuse an implementation from `PointData`? 

The future maintains will be much easier. Feel free to create something like:

```c#
internal static BigInteger? TimeSpanToBigInteger(TimeSpan timestamp, WritePrecision timeUnit)
{
    ...
}
```

Reply:
Sure! Good catch. Since the Builder is a nested class, it could even be private. 
Question here: what should I do with non-covered enum values?
I see 3 alternatives: 
1. return null (and leave BigInteger nullable)
2. throw ArgumentException, like
```cs
switch (timeUnit)
{
   // all cases
default:
   throw new ArgumentOutOfRangeException(nameof(timeUnit), timeUnit, null);
}
```
3. default to one of the cases
For 2. and 3. the return type could be BigInteger (non-nullable)
I am not  big fan of nullables, in my code I use the second option, but I'd like to hear from you so the code is consistent with the rest of the code base.

Reply:
I went with 2. (case for all enum values, default throws). Let me know if it's OK.

Reply:
The `2` is fine. We have to take care to change implementation if there will a new supported precision.